src/csvcubeddevtools/behaviour/csv2rdf.py

```diff
@@ -67,39 +82,82 @@ def _run_csv2rdf(metadata_file_path: Path) -> Tuple[int, str, Optional[str]]:
+            return _run_csv2rdf_internal(tmp_dir)
+    else:
+        return _run_csv2rdf_internal(tmp_dir)
 
 
 @step('csv2rdf on "{file}" should succeed')
 def step_impl(context, file: str):
     """csv2rdf on "{file}" should succeed"""
     temp_dir = get_context_temp_dir_path(context)
-    exit_code, logs, ttl_out = _run_csv2rdf(temp_dir / file)
-    assert exit_code == 0
+    csv2rdf_result = _run_csv2rdf(temp_dir / file)
+    assert csv2rdf_result.status_code == 0, csv2rdf_result.log
+
+    _update_context_for_csv2rdf_result(context, csv2rdf_result)
 
-    context.turtle = ttl_out
+
+def _update_context_for_csv2rdf_result(context, csv2rdf_result: Csv2RdfResult):
+    if hasattr(context, "turtle"):
+        context.turtle += csv2rdf_result.ttl
+    else:
+        context.turtle = csv2rdf_result.ttl
+
+    if not hasattr(context, "rdf_template_data"):
+        context.rdf_template_data = {
+            "rdf_input_directory": _path_to_file_uri_for_csv2rdf_outputs(
+                csv2rdf_result.rdf_input_directory
+            )
+        }
+    else:
+        context.rdf_template_data[
+            "rdf_input_directory"
+        ] = _path_to_file_uri_for_csv2rdf_outputs(csv2rdf_result.rdf_input_directory)
+
+
+def _path_to_file_uri_for_csv2rdf_outputs(file: Path) -> str:
+    """
+    Converts a `pathlib.Path` into a file:/.... URI as output by csv2rdf.
+    """
+
+    file_uri_prefix = "file:" if isinstance(file, PosixPath) else "file:/"
+
+    return file_uri_prefix + os.path.normpath(str(file.absolute())).replace("\\", "/")
 
 
 @step("csv2rdf on all CSV-Ws should succeed")
 def step_impl(context):
     """csv2rdf on all CSV-Ws should succeed"""
-    temp_dir = get_context_temp_dir_path(context)
-    csvw_metadata_files = temp_dir.rglob("*.csv-metadata.json")
+    inputs_temp_dir = get_context_temp_dir_path(context)
+    csvw_metadata_files = inputs_temp_dir.rglob("*.csv-metadata.json")
     context.turtle = ""
 
-    for file in csvw_metadata_files:
-        exit_code, logs, ttl_out = _run_csv2rdf(temp_dir / file)
-        assert exit_code == 0
+    with TemporaryDirectory() as tmp_dir:
+        tmp_dir = Path(tmp_dir)
+
+        for file in csvw_metadata_files:
+            csv2rdf_result = _run_csv2rdf(inputs_temp_dir / file, tmp_dir=tmp_dir)
+            assert csv2rdf_result.status_code == 0, csv2rdf_result.log
 
-        context.turtle += ttl_out
+            _update_context_for_csv2rdf_result(context, csv2rdf_result)
 
 
 @step('csv2rdf on "{file}" should fail with "{expected}"')
 def step_impl(context, file: str, expected: str):
     """csv2rdf on \"{file}\" should fail with \"{expected}\" """
     temp_dir = get_context_temp_dir_path(context)
-    exit_code, logs, ttl_out = _run_csv2rdf(temp_dir / file)
-    assert exit_code == 1
-    assert expected in logs
+    csv2rdf_result = _run_csv2rdf(temp_dir / file)
+    assert csv2rdf_result.status_code != 0
+    assert expected in csv2rdf_result.log, csv2rdf_result.log
```

Comment:
Can we not assert anything specific inside the log? For example, if the csv2rdf is success, do we store a status code in the log? For example, previously we were using `assert expected in logs`